src/otx/algo/common/utils/nms.py

```diff
@@ -39,7 +39,11 @@ def forward(
             valid_mask = scores > score_threshold
             bboxes, scores = bboxes[valid_mask], scores[valid_mask]
             valid_inds = torch.nonzero(valid_mask, as_tuple=False).squeeze(dim=1)
-        inds = torch_nms(bboxes, scores, iou_threshold)
+        if bboxes.get_device() == -1 and bboxes.dtype == torch.bfloat16:  # torch nms kernel doesn't support bfloat16
+            with torch.autocast(device_type="cpu", dtype=torch.bfloat16, enabled=False):
+                inds = torch_nms(bboxes.float(), scores.float(), iou_threshold)
```

Comment:
Is there any way we can throw a message in this case?

Reply:
Looking at the current code, I think the `NMSop` class could hold certain flags (flag for needs of `autocast`) and print the correct message when the first flag changes.

Reply:
> Is there any way we can throw a message in this case?

yes It's not hard, but I'm not sure it is needed. because mixed precision automatically changed precision in available operation by torch originally and nms is not supported operation for CPU. So I think whether autocast is disabled implicitly or explicitly, they are same.

> Looking at the current code, I think the `NMSop` class could hold certain flags (flag for needs of `autocast`) and print the correct message when the first flag changes. What do you think, is this possible?

I'm not sure I understand well what you said, but if you mean need for keeping a flag to decide to autocast or not, I think this thing should be handled by torch side. If misunderstood, please correct me :)